concepts/pivoted/eICU-pivoted-OASIS.sql

```diff
@@ -0,0 +1,571 @@
+  FROM `physionet-data.eicu_crd_derived.pivoted_vital`
+  WHERE (chartoffset > 0 AND chartoffset <= 1440) -- consider only first 24h
+    AND heartrate IS NOT NULL
+    AND patientunitstayid IS NOT NULL 
+  GROUP BY pid_HR, heartrate
+)
+
+-- Mean arterial pressure
+, MAP_OASIS AS (
+  SELECT patientunitstayid AS pid_MAP
+  , CASE
+    WHEN MIN(ibp_mean) < 20.65 THEN 4
+    WHEN MIN(ibp_mean) BETWEEN 20.65 AND 50.99 THEN 3
+    WHEN MIN(ibp_mean) BETWEEN 51 AND 61.32 THEN 2
+    WHEN MIN(ibp_mean) BETWEEN 61.33 AND 143.44 THEN 0
+    WHEN MAX(ibp_mean) >143.44 THEN 3
+    
+    WHEN MIN(nibp_mean) < 20.65 THEN 4
+    WHEN MIN(nibp_mean) BETWEEN 20.65 AND 50.99 THEN 3
+    WHEN MIN(nibp_mean) BETWEEN 51 AND 61.32 THEN 2
+    WHEN MIN(nibp_mean) BETWEEN 61.33 AND 143.44 THEN 0
+    WHEN MAX(nibp_mean) >143.44 THEN 3
+    ELSE NULL
+    END AS MAP_OASIS
+  FROM `physionet-data.eicu_crd_derived.pivoted_vital`
+  WHERE (chartoffset > 0 AND chartoffset <= 1440 ) -- consider only first 24h
+    AND patientunitstayid IS NOT NULL 
+  GROUP BY pid_MAP, ibp_mean, nibp_mean
+)
+
+-- Respiratory rate
+, respiratoryrate_OASIS AS (
+SELECT patientunitstayid AS pid_RR
+  , CASE
+    WHEN MIN(respiratoryrate) < 6 THEN 10
+    WHEN MIN(respiratoryrate) BETWEEN 6 AND 12 THEN 1
+    WHEN MIN(respiratoryrate) BETWEEN 13 AND 22 THEN 0
+    WHEN MAX(respiratoryrate) BETWEEN 23 AND 30 THEN 1
+    WHEN MAX(respiratoryrate) BETWEEN 31 AND 44 THEN 6
+    WHEN MAX(respiratoryrate) > 44 THEN 9
+    ELSE NULL
+    END AS respiratoryrate_OASIS
+
+  FROM `physionet-data.eicu_crd_derived.pivoted_vital`
+  WHERE (chartoffset > 0 AND chartoffset <= 1440 ) -- consider only first 24h
+    AND respiratoryrate IS NOT NULL
+    AND patientunitstayid IS NOT NULL
+  GROUP BY pid_RR, respiratoryrate
+)
+
+-- Temperature 
+, temperature_OASIS AS (
+  SELECT patientunitstayid AS pid_temp
+  , CASE
+    WHEN MIN(temperature) < 33.22 THEN 3
+    WHEN MIN(temperature) BETWEEN 33.22 AND 35.93 THEN 4
+    WHEN MAX(temperature) BETWEEN 33.22 AND 35.93 THEN 4
+    WHEN MIN(temperature) BETWEEN 35.94 AND 36.39 THEN 2
+    WHEN MAX(temperature) BETWEEN 36.40 AND 36.88 THEN 0
+    WHEN MAX(temperature) BETWEEN 36.89 AND 39.88 THEN 2
+    WHEN MAX(temperature) >39.88 THEN 6
+    ELSE NULL
+    END AS temperature_OASIS
+  FROM `physionet-data.eicu_crd_derived.pivoted_vital`
+  WHERE (chartoffset > 0 AND chartoffset <= 1440 ) -- consider only first 24h
+    AND temperature IS NOT NULL
+    AND patientunitstayid IS NOT NULL
+  GROUP BY pid_temp, temperature
+)
+
+-- Urine output
+, urineoutput_OASIS AS (
+  SELECT patientunitstayid AS pid_urine
+  , CASE
+    WHEN MAX(urineoutput) <671 THEN 10
+    WHEN MAX(urineoutput) BETWEEN 671 AND 1426.99 THEN 5
+    WHEN MAX(urineoutput) BETWEEN 1427 AND 2543.99 THEN 1
+    WHEN MAX(urineoutput) BETWEEN 2544 AND 6896 THEN 0
+    WHEN MAX(urineoutput) >6896 THEN 8
+    ELSE NULL
+    END AS urineoutput_OASIS
+
+  FROM `physionet-data.eicu_crd_derived.pivoted_uo`
```

Comment:
This table is slightly different. I am fairly sure you need to SUM the values before aggregating, as the variable is the total urine output over 24 hours, rather than the highest individual measurement of UO.

If you want, you could switch to the `urineoutput` column in the APACHE variables.